src/NEPTypes.jl

```diff
@@ -179,122 +193,121 @@ julia> compute_Mder(nep,1)-(A0+A1*exp(1))
              end
          end
 
+         if !(eltype(AA) <: SparseMatrixCSC)
+             # Dense
+             this=SPMF_NEP_dense{typeof(AA[1]),Ftype}(n,AA,fii,Schur_fact);
+         else
+             # Sparse: Potentially do the joint sparsity pattern trick.
+             T = eltype(AA[1])
+             As = Vector{SparseMatrixCSC{T,Int}}()
+
+             if use_sparsity_pattern && issparse(AA[1])
+                 # Merge the sparsity pattern of all matrices without dropping any zeros
+                 Zero = LinearAlgebra.fillstored!(copy(AA[1]), 1)
+                 for i = 2:size(AA,1)
+                     Zero += LinearAlgebra.fillstored!(copy(AA[i]), 1)
+                 end
+                 Zero = T.(Zero)
+                 Zero.nzval[:] .= T(0)
 
-         T = eltype(AA[1])
-         As = Vector{SparseMatrixCSC{T,Int}}()
-
-         if use_sparsity_pattern && issparse(AA[1])
-             # Merge the sparsity pattern of all matrices without dropping any zeros
-             Zero = LinearAlgebra.fillstored!(copy(AA[1]), 1)
-             for i = 2:size(AA,1)
-                 Zero += LinearAlgebra.fillstored!(copy(AA[i]), 1)
-             end
-             Zero = T.(Zero)
-             Zero.nzval[:] .= T(0)
+                 # Create a copy of each matrix with the sparsity pattern of all matrices combined
+                 @inbounds for A in AA
+                     S = copy(Zero)
 
-             # Create a copy of each matrix with the sparsity pattern of all matrices combined
-             @inbounds for A in AA
-                 S = copy(Zero)
+                     for col = 1:size(A, 2)
+                         for j in nzrange(A, col)
+                             S[A.rowval[j], col] = A.nzval[j]
+                         end
+                     end
 
-                 for col = 1:size(A, 2)
-                    for j in nzrange(A, col)
-                        S[A.rowval[j], col] = A.nzval[j]
-                    end
+                     push!(As, S)
                  end
-
-                 push!(As, S)
+             else
+                 Zero=zeros(n,n)
              end
-         else
-             Zero=zeros(n,n)
-         end
 
+             this=SPMF_NEP_sparse{typeof(AA[1]),Ftype}(n,AA,fii,Schur_fact,Zero,As);
+         end
 
-         this=SPMF_NEP(n,AA,fii,Schur_fact,Zero,As);
          return this
     end
     function SPMF_NEP(n) # Create an empty NEP of size n x n
-         Z=zeros(n,n)
-         return SPMF_NEP(n,Vector{Matrix}(),Vector{Function}(),false,Z);
+        Z=zeros(n,n)
+        return SPMF_NEP_dense{AbstractMatrix}(n,Vector{Matrix}(),Vector{Function}(),false);
```

Comment:
Does this work, with a single type like this?

Reply:
You're right. Fixed.